starter_code/src/movies.js

```diff
@@ -1,15 +1,54 @@
 /* eslint no-restricted-globals: 'off' */
-
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+const orderByYear = (array) => {
+        newArray = array.map(obj => obj)
+        return newArray.sort((a, b) => a.year - b.year || a.title.localeCompare(b.title))
+    }
+    // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+const howManyMovies = array => {
+        let counter = 0;
+        const newArr = array.filter(elem => elem.director === "Steven Spielberg")
+        newArr.forEach(element => element.genre.forEach(ele => {
+            if (ele === "Drama") { counter++ }
+        }))
+        return counter;
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
-
-// Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
-
-// Iteration 4: All rates average - Get the average of all rates with 2 decimals
-
-// Iteration 5: Drama movies - Get the average of Drama Movies
-
-// Iteration 6: Time Format - Turn duration of the movies from hours to minutes
+    }
+    // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+const orderAlphabetically = array => {
+        return newArr = array.map(obj => obj.title).sort().slice(0, 20)
 
-// BONUS Iteration: Best yearly rate average - Best yearly rate average
+    }
+    // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+const ratesAverage = array => {
+        if (array.length > 0) {
+            let acum = 0
+            newArr = array.map(obj => obj.rate)
+            newArr.forEach(element => {
+                if (element) acum += element
+            });
+            return Number(parseFloat(acum / newArr.length).toFixed(2))
+        } else return 0
+    }
+    // Iteration 5: Drama movies - Get the average of Drama Movies
+const dramaMoviesRate = array => {
+        let counter = 0,
+            rate = 0;
```

Comment:
En los iteradores se tiene que definir una condición en la que si el array no contiene ninguna película de Drama devuelva 0, ya que de otra forma está dividiendo counter(0), entre rate (0 ) y 0/0 != 0.

Les recomendamos utilizar **includes()**
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes](url)

Despues de hacer el map para devolver un nuevo array sin mutar el original pueden utilizar in spread operator, de esa manera no se muta el array movies.

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax](url)